ddtrace/tracer/textmap.go

```diff
@@ -1055,6 +1061,11 @@ func parseTracestate(ctx *spanContext, header string) {
 				setPropagatingTag(ctx, "_dd.p."+keySuffix, val)
 			}
 		}
+
+		if !foundP {
+			// indicate that backend could reparent this as a root
+			ctx.reparentID = "0000000000000000"
+		}
```

Comment:
Instead of using `foundP`, checking for it here, and setting a default `ctx.reparentID` value here, you can instead set this value before the loop is entered, and it can just be overwritten the loop if the key is found.
That code in the loop can then become something like this:
```
	else if key == "p" {
		if val != "" {
			ctx.reparentID = val
		}
	}
```

Reply:
Nice, updated to that